protecode/model.py

```diff
@@ -235,48 +279,222 @@ def __hash__(self):
         return hash((self.vulnerability_id(), self.component_name(), self.description()))
 
 
-# --- wrappers for inofficial protecode API responses
-
-
-class ScanResult(ModelBase):
-    def name(self):
-        return self.raw.get('filename', '<None>')
-
-    def is_stale(self) -> bool:
-        '''
-        Returns a boolean value indicating whether or not the stored scan result
-        has become "stale" (meaning that a rescan would potentially return different
-        results).
-        '''
-        return self.raw.get('stale')
-
-    def has_binary(self) -> bool:
-        '''
-        Returns a boolean value indicating whether or not the uploaded file is still present.
-        In case the uploaded file is no longer present, it needs to be re-uploaded prior to
-        rescanning.
-        '''
-        return self.raw.get('rescan-possible')
-
-
 #############################################################################
 ## upload result model
 
-class UploadStatus(Enum):
+class UploadStatus(enum.Enum):
     SKIPPED = 1
     PENDING = 2
     DONE = 4
 
 
 @dataclasses.dataclass
-class BDBA_ScanResult(gcm.ScanResult):
-    # component: cm.Component
-    # resource: cm.Resource
+class BDBA_ScanResult(gcr.ScanResult):
     status: UploadStatus
     result: AnalysisResult
-    licenses: set[License]
-    greatest_cve_score: float = None
 
     @property
-    def license_names(self):
+    def license_names(self) -> typing.Iterable[str]:
         return {l.name() for l in self.licenses}
+
+    @property
+    def licenses(self) -> typing.Sequence[License]:
+        return [c.license() for c in self.result.components() if c.license()]
+
+    @property
+    def greatest_cve_score(self) -> float:
+        return self.result.greatest_cve_score()
+
+
+@dataclasses.dataclass(frozen=True)
+class Binary:
+    def auto_triage(self) -> bool: pass
+    def display_name(self) -> str: pass
+    def metadata(self) -> dict[str, str]:  pass
+    def upload_data(self) -> typing.Generator[bytes, None, None]: pass
+
+
+@dataclasses.dataclass(frozen=True)
+class OciResourceBinary(Binary):
+    artifact: gci.componentmodel.Resource
+
+    def display_name(self):
+        image_reference = self.artifact.access.imageReference
+        _, image_tag = image_reference.split(':')
+        return f'{self.artifact.name}_{image_tag}'
+
+    def upload_data(self) -> typing.Iterable[bytes]:
+        # XXX need to check whether resource is actually a oci-resource
```

Comment:
Should the check be part of this PR?

Reply:
Thinking about it, that case is now handled by Scan-Request-creation - no check necessary, I'll remove the comment